shell/platform/darwin/macos/framework/Source/FLEViewController.mm

```diff
@@ -191,6 +194,12 @@ @implementation FLEViewController {
 static void CommonInit(FLEViewController* controller) {
   controller->_messageHandlers = [[NSMutableDictionary alloc] init];
   controller->_additionalKeyResponders = [[NSMutableOrderedSet alloc] init];
+  [[NSDistributedNotificationCenter defaultCenter]
+      addObserver:controller
+         selector:@selector(onSettingsChanged:)
+             name:@"AppleInterfaceThemeChangedNotification"
+           object:nil];
+  [controller onSettingChanged:nil];
```

Comment:
`init` methods should not call member methods; that's in fact why `CommonInit` is a free function rather than a method. See http://google.github.io/styleguide/objcguide.html#avoid-messaging-the-current-object-within-initializers-and--dealloc

Apart from the subtle issues, this call isn't actually doing anything useful, because _settingsChannel hasn't been created at this point (nor has the engine been started).

What you'll want to do is make a new `sendInitialSettings` (or similar) method, put the `onSettingsChanged:` call there, and call it from `launchEngineInternalWithAssetsPath:...` after the engine has been started successfully.

Reply:
I've moved the sendInitialSettings call to after the engine initialization. This method also contains the initial subscription logic - I'm unsure if this is the correct location.